client/components/editor.vue

```diff
@@ -358,52 +358,28 @@
           this.unconfirmedCount = this.$editorView.state.collab$.unconfirmed.length;
 
           const sendable = collab.sendableSteps(newState);
-          let addingComment = false;
-          let deletingComment = false;
-
           if (sendable) {
-            if (this.commentHighlightKey) {
-              const commentMarks = _.filter(transaction.steps, (s) => {
-                return s.mark && s.mark.type.name === 'highlight' && s.mark.attrs['highlight-keys'] === this.commentHighlightKey;
+            const containsHighlightStep = sendable.steps.find((x) => {
+              return (x.jsonID === 'addMark' || x.jsonID === 'removeMark') && x.mark && x.mark.type.name === 'highlight';
+            });
+            if (this.canUserUpdateDocument || (this.canUserCreateComments && (containsHighlightStep))) {
+              this.lastSentVersion = sendable.version;
+              this.addingStepsInProgress = true;
+              Content.addSteps({
+                contentKey: this.contentKey,
+                currentVersion: sendable.version,
+                steps: sendable.steps.map((step) => {
+                  return step.toJSON();
+                }),
+                clientId: this.clientId,
+              }, (error, response) => {
+                this.addingStepsInProgress = false;
+                if (!error) {
+                  // "content-changed" event is emitted only when addSteps returns without errors.
+                  // This prevents layoutComments from running unnecessarily on the sidebar component.
+                  this.$emit('content-changed');
```

Comment:
So, in fact, if we are precise. Content changes when you get data from the server side, not when method returns, no?

Reply:
Now the 'content-changed' event is emitted when new content is received from server (in pub/sub).